adoc/chapters/programming_interface.adoc

```diff
@@ -2432,7 +2430,8 @@ info::device::version
 
     @ [.code]#std::string#
    a@ Returns the SYCL version as a [code]#std::string# in the form:
-      [code]#<major_version>.<minor_version>#.
+      [code]#<version>.<revision>#, with [code]#<version># being a year based
+      number, for example [code]#2020.4#.
```

Comment:
This query doesn't make sense to me.  The implementation already exposes the SYCL version via the `SYCL_LANGUAGE_VERSION` macro.  Since the entire implementation is conformant to that version, won't every device return the same information via `info::device::version`?

I'm also not sure we should return the minor version number here (or in `SYCL_LANGUAGE_VERSION`).  Do we really expect implementations to change each time we make a minor version release of the spec?  Will we add a conformance test to validate that the implementation returns the most up-to-date minor version?  Will vendors need to re-validate each time there is a new minor release of the spec?

Is it too late to mark this info descriptor as "deprecated"?  We still need to define the semantic even if it is deprecated.  We could either say that it returns "2020", or say something more generic like:

> Returns a vendor-defined string for the device.

Reply:
Perhaps it is a left-over of SYCL 2.2 when we had the idea of having SYCL 1.2 and 2.2 compatible devices in the same program.

Reply:
Updated the PR to mark `info::device::version` as deprecated and make it return just the year based SYCL version.

Reply:
> It would just be a little bit of a shame to "lose" a simple property like that, that could easily be used by backends to report more informative information on the device such as OpenCL device version, architecture, compute capabilities and such.

If you feel this way, then we should not add the wording you proposed in 2ecbb672466a08e267ab32c1ef0d0385119044dc.  Instead, we should say something like:

> Returns a vendor-defined string for the device

This gives vendor's the flexibility to return information like you suggest.

Reply:
Updated with what I suggested, although, note that I was thinking of backend-defined rather than vendor-defined, so what the value should be would still be defined by the SYCL spec but just on a per backend basis.

I feel like that gives us enough flexibility to report useful information for the specific backend types while still giving some consistency across vendors.